win32ss/user/user32/windows/dialog.c

```diff
@@ -2624,7 +2624,17 @@ IsDialogMessageW(
              if (!(dlgCode & DLGC_WANTARROWS))
              {
                  BOOL fPrevious = (lpMsg->wParam == VK_LEFT || lpMsg->wParam == VK_UP);
-                 HWND hwndNext = GetNextDlgGroupItem( hDlg, lpMsg->hwnd, fPrevious );
+                 HWND hwndFirst = lpMsg->hwnd;
+                 HWND hwndNext = GetNextDlgGroupItem(hDlg, hwndFirst, fPrevious);
+
+                 /* Skip STATIC elements when arrow-moving through a list of controls */
+                 while (hwndNext && (SendMessageW(hwndNext, WM_GETDLGCODE, 0, 0) & DLGC_STATIC))
+                 {
+                     if (hwndFirst == hwndNext)
+                         return TRUE;
+                     hwndNext = GetNextDlgGroupItem(hDlg, hwndNext, fPrevious);
+                 }
```

Comment:
Is the following suggestion more or less equivalent to the proposed code? But it always breaks the loop and allow it to continue to the code below. Except, indeed, that if we get the hwndNext to be == hwndFirst (and not NULL), it would probably take the first if-branch of the code below and send an undesirable WM_GETDLGCODE .... ?
```suggestion

                 /* Skip STATIC elements when arrow-moving through a list of controls */
                 HWND hwndNext, hwndFirst = lpMsg->hwnd;
                 for (hwndNext = GetNextDlgGroupItem(hDlg, hwndFirst, fPrevious);
                      hwndNext && hwndFirst != hwndNext;
                      hwndNext = GetNextDlgGroupItem(hDlg, hwndNext, fPrevious))
                  {
                      if (!(SendMessageW(hwndNext, WM_GETDLGCODE, 0, 0) & DLGC_STATIC))
                          break;
                  }
```

Reply:
Thanks.